packages/ti_stix/data_stream/indicator/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,371 @@
+        ctx._conf?.restrict_stix != null &&
+        ctx._conf.restrict_stix == true
+      tag: drop_no_indicator
+  - drop:
+      if: >-
+        ctx.stix?.spec_version != '2.1' &&
+        ctx._conf?.restrict_stix != null &&
+        ctx._conf.restrict_stix == true
+      tag: drop_no_stix_2_1
+  - date:
+      field: stix.created
+      target_field: "@timestamp"
+      formats:
+        - "ISO8601"
+      if: ctx.stix?.created != null
+      tag: date_created
+  - date:
+      field: stix.modified
+      target_field: threat.indicator.last_seen
+      formats:
+        - ISO8601
+      if: ctx.stix?.modified != null
+      tag: date_modified
+  - set:
+      field: threat.indicator.modified_at
+      copy_from: threat.indicator.last_seen
+      if: ctx.threat?.indicator?.last_seen != null
+      tag: set_threat_indicator_last_seen
+  - date:
+      field: stix.valid_from
+      target_field: threat.indicator.first_seen
+      formats:
+        - "ISO8601"
+      if: ctx.stix?.valid_from != null
+      tag: date_valid_from
+  - rename:
+      field: stix.name
+      target_field: threat.indicator.name
+      ignore_missing: true
+      tag: rename_name
+  - rename:
+      field: stix.description
+      target_field: threat.indicator.description
+      ignore_missing: true
+      tag: rename_description
+  - script:
+      lang: painless
+      if: ctx.stix?.confidence != null
+      description: Normalize confidence level.
+      tag: script_normalize_confidence
+      source: >
+        def value = ctx.stix.confidence;
+        if (value == 0) {
+          ctx.threat.indicator.confidence = "None";
+        } else if (value >= 1 && value <= 29) {
+          ctx.threat.indicator.confidence = "Low";
+        } else if (value >= 30 && value <= 69) {
+          ctx.threat.indicator.confidence = "Medium";
+        } else if (value >= 70 && value <= 100) {
+          ctx.threat.indicator.confidence = "High";
+        } else {
+          ctx.threat.indicator.confidence = "Not Specified";
+        }
+  - foreach:
+      field: stix.labels
+      if: ctx.stix?.labels != null
+      processor:
+        append:
+          field: tags
+          value: '{{ _ingest._value }}'
+          tag: append_tags
+      ignore_missing: true
+      ignore_failure: true
+  - remove:
+      field: stix.labels
+      ignore_missing: true
+
+  ## IOC expiration
+  - date:
+      field: stix.valid_until
+      target_field: stix.ioc_expiration_date
+      formats:
+        - ISO8601
+      if: ctx.stix?.valid_until != null
+      tag: date_ioc_expiration_valid_until
+  - set:
+      field: stix.ioc_expiration_date
+      copy_from: threat.indicator.modified_at
+      if: >-
+        ctx.stix?.ioc_expiration_date == null &&
+        ctx.stix?.revoked != null &&
+        ctx.stix.revoked == true
+      tag: set_ioc_expiration_revoked
+
+  - script:
+      lang: painless
+      tag: script_conf_ioc_expiration
+      if: >-
+        ctx.stix.ioc_expiration_date == null &&
```

```suggestion
        ctx.stix?.ioc_expiration_date == null &&
```
?

Mainly for consistency; there are expressions above that suggest that ctx.stix could be null, though this should never be the case unless the json processor fails, in which case we are lost anyway. The alternative is to `s/ctx\.stix\?/ctx.stix/g`, maybe.